hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SStruct.scala

```diff
@@ -23,9 +24,15 @@ trait SBaseStructValue extends SValue {
   def loadField(cb: EmitCodeBuilder, fieldName: String): IEmitSCode = loadField(cb, pt.fieldIdx(fieldName))
 }
 
-trait SBaseStructCode extends SCode {
+trait SBaseStructCode extends SCode { self =>
+  def st: SStruct
+
   def memoize(cb: EmitCodeBuilder, name: String): SBaseStructValue
 
   def memoizeField(cb: EmitCodeBuilder, name: String): SBaseStructValue
-}
 
+  def subset(fieldNames: String*): SSubsetStructCode = {
+    val st = SSubsetStruct(self.st, fieldNames.toIndexedSeq)
```

Comment:
match on `st` and fail if it's a tuple, maybe?

Reply:
There is no STuple.

Reply:
right, I mean match on the virtual type

Reply:
Oops, haven't PRed that yet. You can add a `def virtualType: Type = pType.virtualType` to the `sType` trait.

Reply:
I don't see the purpose of such a check.

Reply:
OK, I think you're right that we don't need this right now. I'll keep an eye out for usages that might motivate this check though.